internal/bminventory/inventory.go

```diff
@@ -4777,8 +4883,430 @@ func (b *bareMetalInventory) AddOpenshiftVersion(ctx context.Context, ocpRelease
+			UpdatedAt:              strfmt.DateTime{},
+			AdditionalNtpSources:   swag.StringValue(params.InfraenvCreateParams.AdditionalNtpSources),
+			SSHAuthorizedKey:       swag.StringValue(params.InfraenvCreateParams.SSHAuthorizedKey),
+		},
+		KubeKeyNamespace: kubeKey.Namespace,
+	}
+
+	if params.InfraenvCreateParams.ClusterID != nil {
+		infraEnv.ClusterID = *params.InfraenvCreateParams.ClusterID
+	}
+	if params.InfraenvCreateParams.Proxy != nil {
+		proxy := models.Proxy{
+			HTTPProxy:  params.InfraenvCreateParams.Proxy.HTTPProxy,
+			HTTPSProxy: params.InfraenvCreateParams.Proxy.HTTPSProxy,
+			NoProxy:    params.InfraenvCreateParams.Proxy.NoProxy,
+		}
+		infraEnv.Proxy = &proxy
+		var infraEnvProxyHash string
+		infraEnvProxyHash, err = computeProxyHash(&proxy)
+		if err != nil {
+			msg := "Failed to compute infraEnv proxy hash"
+			log.Error(msg, err)
+			return nil, common.NewApiError(http.StatusInternalServerError, errors.New(msg))
+		}
+		infraEnv.ProxyHash = infraEnvProxyHash
+	}
+
+	pullSecret := swag.StringValue(params.InfraenvCreateParams.PullSecret)
+	err = b.secretValidator.ValidatePullSecret(pullSecret, ocm.UserNameFromContext(ctx), b.authHandler)
+	if err != nil {
+		err = errors.Wrap(secretValidationToUserError(err), "pull secret for new infraEnv is invalid")
+		return nil, common.NewApiError(http.StatusBadRequest, err)
+	}
+	ps, err := b.updatePullSecret(pullSecret, log)
+	if err != nil {
+		return nil, common.NewApiError(http.StatusBadRequest,
+			errors.New("Failed to update Pull-secret with additional credentials"))
+	}
+	setInfraEnvPullSecret(&infraEnv, ps)
+
+	if err = updateSSHAuthorizedKey(&infraEnv); err != nil {
+		return nil, common.NewApiError(http.StatusBadRequest, err)
+	}
+
+	if err = b.db.Create(&infraEnv).Error; err != nil {
+		return nil, common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	if err = b.GenerateInfraEnvISOInternal(ctx, &infraEnv); err != nil {
+		return nil, common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	success = true
+	return b.GetInfraEnvInternal(ctx, installer.GetInfraEnvParams{InfraEnvID: infraEnv.ID})
+}
+
+func (b *bareMetalInventory) UpdateInfraEnv(ctx context.Context, params installer.UpdateInfraEnvParams) middleware.Responder {
+	i, err := b.updateInfraEnvInternal(ctx, params)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	return installer.NewUpdateInfraEnvCreated().WithPayload(&i.InfraEnv)
+}
+
+func (b *bareMetalInventory) updateInfraEnvInternal(ctx context.Context, params installer.UpdateInfraEnvParams) (*common.InfraEnv, error) {
+	log := logutil.FromContext(ctx, b.log)
+	var infraEnv *common.InfraEnv
+	var err error
+	log.Infof("update infraEnv %s with params: %+v", params.InfraEnvID, params.InfraEnvUpdateParams)
+
+	if params, err = b.validateAndUpdateInfraEnvParams(ctx, &params); err != nil {
+		return nil, common.NewApiError(http.StatusBadRequest, err)
+	}
+
+	txSuccess := false
+	tx := b.db.Begin()
```

Comment:
why do we need transaction here? we are updating only one entry in the one DB table, no?

Reply:
removing it